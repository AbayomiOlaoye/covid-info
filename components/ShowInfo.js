import { useLocation, NavLink } from 'react-router-dom';

import React from 'react';

  <NavLink details to="/">
    <h1>Go</h1>
  </NavLink>;

const ShowInfo = () => {
  const location = useLocation();
  const { state } = location;

  return (
    <div>
      <div className="main">
        <h1>{state.country}</h1>
        <p>{state.updated}</p>
      </div>

      <div className="self_display">
        <h1>Live Information</h1>
        <p>
          {' '}
          Continent
          {' '}
          <span>
            {state.continent}
            {' '}
          </span>
        </p>
        <p>
          {' '}
          Cases
          {' '}
          <span>
            {state.cases}
            {' '}
          </span>
        </p>
        <p>
          {' '}
          Today Case
          {' '}
          <span>
            {state.todayCases}
            {' '}
          </span>
        </p>
        <p>
          {' '}
          Death
          {' '}
          <span>
            {state.deaths}
            {' '}
          </span>
        </p>
        <p>
          {' '}
          Today Death
          {' '}
          <span>
            {state.todayDeaths}
            {' '}
          </span>
        </p>
        <p>
          {' '}
          Recovered
          {' '}
          <span>
            {state.recovered}
            {' '}
          </span>
        </p>
        <p>
          {' '}
          Today Recovered
          {' '}
          <span>
            {state.todayRecovered}
            {' '}
          </span>
        </p>
        <p>
          {' '}
          Tests
          {' '}
          <span>
            {state.tests}
            {' '}
          </span>
        </p>
      </div>
    </div>
  );
};

export default ShowInfo;
