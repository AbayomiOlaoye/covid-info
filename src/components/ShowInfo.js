import { useLocation } from 'react-router-dom';
import React from 'react';

const ShowInfo = () => {
  const location = useLocation();
  const { state } = location;

  return (
    <>
      <div className="infoCard">
        <h1 className="upDate info-head">Live Information</h1>
        <table className="table-data">
          <tr>
            <th>Continent:</th>
            <td>{state.continent}</td>
          </tr>
          <tr>
            <th>Cases:</th>
            <td>{state.cases}</td>
          </tr>
          <tr>
            <th>Today&apos;s Case:</th>
            <td>{state.todayCases}</td>
          </tr>
          <tr>
            <th>Deaths:</th>
            <td>{state.deaths}</td>
          </tr>
          <tr>
            <th>Total Deaths:</th>
            <td>{state.todayDeaths}</td>
          </tr>
          <tr>
            <th>Recovered Cases Info:</th>
            <td>{state.recovered}</td>
          </tr>
          <tr>
            <th>Today&apos;s Recovery Info:</th>
            <td>{state.todayRecovered}</td>
          </tr>
          <tr>
            <th>Tests:</th>
            <td>{state.tests}</td>
          </tr>
        </table>
      </div>
    </>
  );
};

export default ShowInfo;
