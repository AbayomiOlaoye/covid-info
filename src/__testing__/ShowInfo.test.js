import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import ShowInfo from '../components/ShowInfo';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    state: {
      continent: 'Africa',
      cases: '16',
      todayCases: '123',
      deaths: '25',
      todayDeaths: '44',
      recovered: '23',
      todayRecovered: '34',
      test: '123',
    },
  }),
}));

test('Check if details page renders correctly', () => {
  const tree = render(
    <Router>
      <ShowInfo />
    </Router>,
  );
  expect(tree).toMatchSnapshot();
});
