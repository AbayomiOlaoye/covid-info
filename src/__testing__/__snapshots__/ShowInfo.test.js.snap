// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['Check if details page renders correctly 1'] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="infoCard"
      >
        <h1
          class="upDate info-head"
        >
          Live Information
        </h1>
        <table
          class="table-data"
        >
          <tr>
            <th>
              Continent:
            </th>
            <td>
              Africa
            </td>
          </tr>
          <tr>
            <th>
              Cases:
            </th>
            <td>
              16
            </td>
          </tr>
          <tr>
            <th>
              Today's Case:
            </th>
            <td>
              123
            </td>
          </tr>
          <tr>
            <th>
              Deaths:
            </th>
            <td>
              25
            </td>
          </tr>
          <tr>
            <th>
              Total Deaths:
            </th>
            <td>
              44
            </td>
          </tr>
          <tr>
            <th>
              Recovered Cases Info:
            </th>
            <td>
              23
            </td>
          </tr>
          <tr>
            <th>
              Today's Recovery Info:
            </th>
            <td>
              34
            </td>
          </tr>
          <tr>
            <th>
              Tests:
            </th>
            <td />
          </tr>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="infoCard"
    >
      <h1
        class="upDate info-head"
      >
        Live Information
      </h1>
      <table
        class="table-data"
      >
        <tr>
          <th>
            Continent:
          </th>
          <td>
            Africa
          </td>
        </tr>
        <tr>
          <th>
            Cases:
          </th>
          <td>
            16
          </td>
        </tr>
        <tr>
          <th>
            Today's Case:
          </th>
          <td>
            123
          </td>
        </tr>
        <tr>
          <th>
            Deaths:
          </th>
          <td>
            25
          </td>
        </tr>
        <tr>
          <th>
            Total Deaths:
          </th>
          <td>
            44
          </td>
        </tr>
        <tr>
          <th>
            Recovered Cases Info:
          </th>
          <td>
            23
          </td>
        </tr>
        <tr>
          <th>
            Today's Recovery Info:
          </th>
          <td>
            34
          </td>
        </tr>
        <tr>
          <th>
            Tests:
          </th>
          <td />
        </tr>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
